src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -53,5 +294,31 @@ protected static ILogger CreateLogger<T>(ILoggerFactory factory)
 
             return factory.CreateLogger<T>();
         }
+
+        protected static async Task WriteFileAsync(HttpContext context, Stream fileStream, RangeItemHeaderValue range, long rangeLength)
+        {
+            var outputStream = context.Response.Body;
+            using (fileStream)
+            {
+                try
+                {
+                    if (range == null)
+                    {
+                        await StreamCopyOperation.CopyToAsync(fileStream, outputStream, count: null, bufferSize: BufferSize, cancel: context.RequestAborted);
+                    }
+                    else
+                    {
+                        fileStream.Seek(range.From.Value, SeekOrigin.Begin);
```

Comment:
For all file types except the `FileStream`, the stream is retrieved right before I call this method (in order to write to it). If I want to not set the headers for CanSeek = false streams, I should move the code to get the stream in `ExecuteAsync` before calling `SetHeadersAndLog` and then send that in to the WriteFileAsync